### PR TITLE
Changes /obj/machinery to have atom/movable occupants

### PR DESCRIPTION
**Pull request by coiax**
Currently `occupant` in /obj/machinery is a /mob/living type. I have
changed it to /atom/movable, and made a type cache to determine what to
"look for" when `close_machine()` is called.

This means that machines can have non-mob occupants. I have some design
ideas for this, but that's after the freeze.

### CODE REVIEW

**Review by Jalleo — CHANGES_REQUESTED** [review]
I approve of these changes if there arent any bugs discovered.

**Comment by xxalpha** [comment]
This is like a feature almost?

File: code/game/machinery/Sleeper.dm
Comment (RemieRichards) [inline code/game/machinery/Sleeper.dm]: this does not guarantee they are a mob.

File: code/game/machinery/Sleeper.dm
Comment (RemieRichards) [inline code/game/machinery/Sleeper.dm]: ```istype(mob_occupant) && ...```

File: code/game/machinery/Sleeper.dm
Comment (RemieRichards) [inline code/game/machinery/Sleeper.dm]: actually, why does most of the code do isliving checks? shouldn't the typecache be enough? if admins break it, fuck admins.

Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/Sleeper.dm]: fuck you shitty admin

**Comment by duncathan** [comment]
@coiax

**Comment by MrPerson** [comment]
Can I ask what kind of stuff you were intending?

**Comment by coiax** [comment]
Put brains and heads in the cloning scanner.

**Comment by optimumtact** [comment]
are you planning on responding to any of the maintainer comments?

File: code/game/machinery/Sleeper.dm
Comment (coiax) [inline code/game/machinery/Sleeper.dm]: so Remie, should I do type checking constantly, or just assume that we're not badminned?

Comment (RemieRichards) [inline code/game/machinery/Sleeper.dm]: assume not badminned.

**Comment by coiax** [comment]
I'm working on removing the checks currently.

**Comment by coiax** [comment]
@RemieRichards I have removed all the checks. Also I made VR pods human only.

**Comment by optimumtact** [comment]
conflicted

**Comment by optimumtact** [comment]
rip conflicting again sorry

**Comment by duncathan** [comment]
will reopen when no longer stale

**Comment by duncathan** [comment]
JK LOL

**Comment by MrStonedOne** [comment]
This pr increased the server's memeory usage on box station by 220MB

**Comment by duncathan** [comment]
oops

**Comment by optimumtact** [comment]
thanks for the update sport

**Pull request merged**
